### PR TITLE
fix(ivy): support attribute selectors in dynamic component creation

### LINKED ISSUE #31785: Regression Ivy with ComponentFactory
# 🐞 bug report

Tbh: I didn't search if this was already known (no idea how I would have found it 😅)

### Affected Package
The issue is caused by package @angular/core

### Is this a regression?

This works without Ivy enabled.

### Description

I was trying out Angular Ivy in our codebase. One of the last 'problems' that is really blocking seems to be ['ngx-toastr'](https://github.com/scttcper/ngx-toastr).

They dynamically add the [toastr component](https://github.com/scttcper/ngx-toastr/blob/master/src/lib/toastr/toast.component.ts) on runtime in the DOM.

[Link to code that is responsible for this](https://github.com/scttcper/ngx-toastr/blob/db3e8fefe79dd66f2610b057a762bb0e668ab417/src/lib/portal/dom-portal-host.ts#L32)

When switching to Ivy this code doesn't work anymore:

```
ERROR DOMException: Failed to execute 'createElement' on 'Document': The tag name provided ('') is not a valid name. at DefaultDomRenderer2.createElement
```

After some debugging I found out that it's because the code that's responsible for retrieving the selector of a component in Ivy doesn't work: [Link](https://github.com/angular/angular/blob/c845a7b887d1a85a89acfe5c62b8829079130af3/packages/core/src/render3/component_ref.ts#L124)

In case of the toast component (see link above) it's ['', 'toast-component', '']. As a result the tag name becomes ''. And that explains the error above.

I cloned ngx-toastr, changed the component name from `[toast-component]` to `toast-component` (removed the brackets). And everything seems to be working. This of course changes the result from a div with as attribute 'toast-component' to a custom component with as selector 'toast-component'.

TL;DR: Dynamically generating components on runtime with Ivy doesn't work when using attribute notation.

## 🔬 Minimal Reproduction

Use ngx-toastr with Ivy enabled.
If more time, I'll create a stackblitz.

## 🔥 Exception or Error

```
ERROR DOMException: Failed to execute 'createElement' on 'Document': The tag name provided ('') is not a valid name. 
at DefaultDomRenderer2.createElement (http://localhost:4200/vendor.js:82322:25) 
at elementCreate (http://localhost:4200/vendor.js:53703:32) 
at ComponentFactory$1.create (http://localhost:4200/vendor.js:67815:13) 
at DomPortalHost.attachComponentPortal (http://localhost:4200/vendor.js:94547:41) 
at DomPortalHost.attach (http://localhost:4200/vendor.js:94505:21) at OverlayRef.attach (http://localhost:4200/vendor.js:94629:33) 
at ToastrService._buildNotification (http://localhost:4200/vendor.js:94955:35) 
at ToastrService._preBuildNotification (http://localhost:4200/vendor.js:94921:21) 
at ToastrService.success
```

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 8.2.0-next.1
Node: 12.6.0
OS: win32 x64
Angular:
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.802.0-next.1
@angular-devkit/core         8.2.0-next.1
@angular-devkit/schematics   8.2.0-next.1
@schematics/angular          8.2.0-next.1
@schematics/update           0.802.0-next.1
rxjs                         6.4.0

</code></pre>

**Anything else relevant?**
Causes https://github.com/scttcper/ngx-toastr/issues/686

### PR DESCRIPTION
**Pull request by kara**
This commit fixes a bug where we assumed all dynamically created
components would have tag-name selectors, so we passed through the
"tag name" to the renderer as the first index of the selector. For
components with attribute selectors, the tag name would be "", so
the renderer would try to create an element with tag name "" and
throw. Now we default to a "div" element the same way that View
Engine did.

Closes #31785

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍

**Comment by kara** [comment]
[VE presubmit](https://test.corp.google.com/ui#id=OCL:259637650:BASE:259637662:1563925146992:b1cd0f2b)

[Ivy presubmit](http://test/OCL:259637650:BASE:259638762:1563925493734:9b96a7ed)

**Comment by kara** [comment]
merge-assistance: code fresh is hanging
